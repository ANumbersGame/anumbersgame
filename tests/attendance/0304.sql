use DebateResults0304;

alter table MasterTournaments
drop column DistrictQual;

alter table MasterTournaments
add column DistrictQual
int null
default null
comment 'Not null if tournament is district NDT qualifier';

update MasterTournaments
set DistrictQual = 1
where TourneyName = 'NDT D1 Qualifying and CA Junior-Novice Champs';

update MasterTournaments
set DistrictQual = 3
where TourneyName = 'D3 NDT Qualifying tournament';

update MasterTournaments
set DistrictQual = 4
where TourneyName = 'District IV NDT Qualifier';

update MasterTournaments
set DistrictQual = 5
where TourneyName = 'District 5 Qualifier';

update MasterTournaments
set DistrictQual = 6
where TourneyName = 'SECEDA/SCCEDA/D-6';

update MasterTournaments
set DistrictQual = 7
where TourneyName = 'District VII Qualifying Tournament';

update MasterTournaments
set DistrictQual = 8
where TourneyName = 'NDT District 8';

update MasterTournaments
set DistrictQual = 9
where TourneyName = 'District 9 NDT Qualifier';

select count(*) = 8
from MasterTournaments
where DistrictQual is not null;

select DistrictQual as Dist, count(*) as Rounds, Diivision as 'Div', TourneyName
from MasterResults, MasterTournaments
where Tournament = MasterTournaments.ID
and DistrictQual is not null
group by Tournament, Diivision
order by Diivision, DistrictQual;


select count(distinct AffTeam) as 'District Teams'
from MasterResults, MasterTournaments
where Tournament = MasterTournaments.ID
and (DistrictQual = 1
or DistrictQual = 3
or DistrictQual = 5
or DistrictQual = 6
or DistrictQual = 7
or DistrictQual = 8)
and Diivision = 1;

select count(distinct AffTeam) as 'CEDA Teams'
from MasterResults, MasterTournaments
where Tournament = MasterTournaments.ID
and TourneyName = 'CEDA Nationals';