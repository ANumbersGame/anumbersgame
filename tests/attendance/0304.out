count(*) = 8
1
Dist	Rounds	Div	TourneyName
1	30	1	NDT D1 Qualifying and CA Junior-Novice Champs
3	73	1	D3 NDT Qualifying tournament
4	12	1	District IV NDT Qualifier
5	48	1	District 5 Qualifier
6	73	1	SECEDA/SCCEDA/D-6
7	72	1	District VII Qualifying Tournament
8	51	1	NDT District 8
1	49	2	NDT D1 Qualifying and CA Junior-Novice Champs
6	19	2	SECEDA/SCCEDA/D-6
1	55	3	NDT D1 Qualifying and CA Junior-Novice Champs
6	33	3	SECEDA/SCCEDA/D-6
District Teams
95
CEDA Teams
174
